server/coordinator/procedure/operation/split/split.go

```diff
@@ -113,31 +144,43 @@ func (p *Procedure) Typ() procedure.Typ {
 	return procedure.Split
 }
 
+func (p *Procedure) RelatedVersionInfo() procedure.RelatedVersionInfo {
+	shardWithVersion := make(map[storage.ShardID]uint64, 0)
+	for _, shardView := range p.snapshot.Topology.ShardViews {
+		if shardView.ShardID == p.shardID {
+			shardWithVersion[p.shardID] = shardView.Version
+		}
+	}
+	shardWithVersion[p.newShardID] = 0
+	return procedure.RelatedVersionInfo{
+		ClusterID:        p.snapshot.Topology.ClusterView.ClusterID,
+		ShardWithVersion: shardWithVersion,
+		ClusterVersion:   p.snapshot.Topology.ClusterView.Version,
+	}
```

Comment:
It'll be better to store it as the field of Procedure if the method will be called multiple times.